sql/updates/world/3.3.5/9999_99_99_99_world.sql

```diff
@@ -0,0 +1,85 @@
+
+-- TODO 22419 wood mites do not attack player on spawn. Not sure how to handle this in practice. Proposed fix:
+-- Wood Mite
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 22419;
+
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 22419);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(22419, 0, 0, 0, 54, 0, 100, 0, 0, 0, 0, 0, 49, 0, 0, 0, 0, 0, 0, 21, 15, 0, 0, 0, 0, 0, 0, 'Wood Mite - On Just Summoned - Start Attacking');
+
+-- Skull Pile
+UPDATE `gameobject_template` SET `ScriptName` = '', `AIName` = 'SmartGameObjectAI' WHERE `entry` = 185913;
+DELETE FROM `smart_scripts` WHERE `entryorguid` = 185913 AND `source_type` = 1;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(185913, 1, 0, 1, 62, 0, 100, 0, 8660, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 0 Selected - Close Gossip'),
+(185913, 1, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 85, 40632, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 0 Selected - Invoker Cast \'Summon Gezzarak the Huntress\''),
+(185913, 1, 2, 3, 62, 0, 100, 0, 8660, 1, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 1 Selected - Close Gossip'),
+(185913, 1, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 85, 40640, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 1 Selected - Invoker Cast \'Summon Karrog\''),
+(185913, 1, 4, 5, 62, 0, 100, 0, 8660, 2, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 2 Selected - Close Gossip'),
+(185913, 1, 5, 0, 61, 0, 100, 0, 0, 0, 0, 0, 85, 40642, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 2 Selected - Invoker Cast \'Summon Darkscreecher Akkarai\''),
+(185913, 1, 6, 7, 62, 0, 100, 0, 8660, 3, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 3 Selected - Close Gossip'),
+(185913, 1, 7, 0, 61, 0, 100, 0, 0, 0, 0, 0, 85, 40644, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 3 Selected - Invoker Cast \'Summon Vakkiz the Windrager\'');
```

Comment:
> Gossip does not close properly on Skull Pile (GO 185913) despite event linking working fine. Would appreciate dev eyes, I can't figure it out

Try to close gossip only after spellcast. Actually it's standart for such scripts and maybe will fix the issue.

Reply:
this didn't seem to fix it, it just closed the gossip w/o invoker casting. there are no conditions in my DB that would cause this so I'm at a loss

Reply:
fwiw, I tried

```sql
UPDATE `gameobject_template` SET `ScriptName` = '', `AIName` = 'SmartGameObjectAI' WHERE `entry` = 185913;
DELETE FROM `smart_scripts` WHERE `entryorguid` = 185913 AND `source_type` = 1;
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(185913, 1, 0, 4, 62, 0, 100, 0, 8660, 0, 0, 0, 85, 40632, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 0 Selected - Invoker Cast \'Summon Gezzarak the Huntress\''),
(185913, 1, 1, 4, 62, 0, 100, 0, 8660, 1, 0, 0, 85, 40640, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 1 Selected - Invoker Cast \'Summon Karrog\''),
(185913, 1, 2, 4, 62, 0, 100, 0, 8660, 2, 0, 0, 85, 40642, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 2 Selected - Invoker Cast \'Summon Darkscreecher Akkarai\''),
(185913, 1, 3, 4, 62, 0, 100, 0, 8660, 3, 0, 0, 85, 40644, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 3 Selected - Invoker Cast \'Summon Vakkiz the Windrager\''),
(185913, 1, 4, 0, 61, 0, 100, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Skull Pile - On Gossip Option 0 Selected - Close Gossip');

Reply:
OK, so that means to swap the order of the 2 rows above, I guess?

Reply:
well, what's in the PR currently is "close gossip, then cast spell". gossip does not close, but spell casts.

the above script reverses that order in general, so "cast spell, then close gossip". Now gossip closes but spell does not cast.

Reply:
Strange, there's a lot of similar scripts. Can't believe they don't work.

Reply:
Maybe event_param2 (Cast Flags) aren't alright

Reply:
mm maybe? but if you reverse the order (have the cast as the linked event) it casts properly

Reply:
actually yeah castflags 4 fixes it, but I dont know if thats acceptable

Reply:
I'm having trouble reproing it not working now, actually.

Reply:
I remember some of my SAI scripts (a year back ++) often had a tendency to not work for me, but work OK for others (at least during writing and testing).

Reply:
`.reload smart_scripts` didn't work properly. I had to unload the grid and it worked.